js/AbstractPropertiesList.js

```diff
@@ -0,0 +1,157 @@
+/**
+ * Copyright 2017-present, The Visdom Authors
+ * All rights reserved.
+ *
+ * This source code is licensed under the license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import React from 'react';
+
+class MyRef {
+  constructor() {
+    this._ref = null;
+  }
+
+  getRef() {
+    return this._ref;
+  }
+
+  setRef = ref => {
+    this._ref = ref;
+  };
+}
+
+class Text extends React.Component {
```

Comment:
Can we rename this class while we're moving it? `Text` was fairly nondescript before, perhaps `EditablePropertyText`?

Reply:
Sure, I adapted the code accordingly.